app/controllers/DayAggregateEventController.scala

```diff
@@ -16,40 +16,45 @@
 
 package controllers
 
+import model.persisted.eventschedules.Location
 import org.joda.time.LocalDate
-import play.api.libs.json.{ Json, OFormat }
-import play.api.mvc.{ Action, AnyContent }
-import repositories.events.EventsRepository
+import play.api.libs.json.{Json, OFormat}
+import play.api.mvc.{Action, AnyContent}
+import repositories.events.{EventsRepository, LocationsWithVenuesRepository, LocationsWithVenuesYamlRepository}
 import uk.gov.hmrc.play.microservice.controller.BaseController
 
 import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
 object DayAggregateEventController extends DayAggregateEventController {
-  override val eventsRepository: EventsRepository = repositories.eventsRepository
+  val locationsWithVenuesRepo = LocationsWithVenuesYamlRepository
+  val eventsRepository: EventsRepository = repositories.eventsRepository
 }
 
 trait DayAggregateEventController extends BaseController {
+  def locationsWithVenuesRepo: LocationsWithVenuesRepository
   def eventsRepository: EventsRepository
 
   def findBySkillTypes(skillTypes: String): Action[AnyContent] = Action.async { implicit request =>
-    find(Some(skillTypes), None)
+    find(Some(skillTypes), None).map ( dayAggregateEvents => Ok(Json.toJson(dayAggregateEvents)) )
   }
 
   def findBySkillTypesAndLocation(skillTypes: String, location: String): Action[AnyContent] = Action.async { implicit request =>
-    find(Some(skillTypes), Some(location))
+    locationsWithVenuesRepo.location(location).flatMap { location =>
```

Comment:
We would still need to create a Location and Venue case class in our tests so we don't rely on the repo but I'm not sure how to prevent it in the app code but not in the tests?